### DISCUSSION TITLE
ESRI to EPSG

### DISCUSSION THREAD

**Discussion by hqnghi88** [gama-platform/gama#132] [Q&A]
Dear all,
I am working with a dataset which is projected in ESRI:53004. With QGIS i can use the export layer to save as EPSG:3857 (or else), but i can not succed within GAMA to do the same job. Does anyone have an idea or some solution? I tried with to_GAMA_CRS (no message but the convert seem not correct) and CRS_transform (with an error message of not found ESRI).
Thanks.

**Comment by ptaillandier** [comment]
Indeed, GAMA does not seem to understand CRS based on ESRI code. When I tried, the geometries are well exported with the good size, but not at the right location (after opening the shapefile in QGIS). You can add an issue about it, even it is not a priority for me to solve this issue (as a workaround is to use QGIS, and as 95% of shapefiles use a CRS based on EPSG code).

By the way, the simplest way to achieved what you want is to use this code:

```
global {
	shape_file shape_file_buildings <- shape_file("../includes/buildings_simple.shp");
	geometry shape <- envelope(shape_file_buildings);
	
	init {
		save shape_file_buildings.contents format:"shp" to:"building.shp" crs:"EPSG:3857" ;
	}
}
```

**Comment by hqnghi88** [comment]
Yes, it have always a misplaced location, i tried also with save statement (w/wo .contents) and from the species. It's ok for workaround in qgis, just un-sure about the capabilities as qgis also based on geotools library.
Thank you anyway.
